packages/concerto-core/lib/introspect/stringvalidator.js

```diff
@@ -70,7 +70,7 @@ class StringValidator extends Validator{
                 this.regex = new CustomRegExp(validator.pattern, validator.flags);
             }
             catch(exception) {
-                this.reportError(field.getName(), exception.message);
+                this.reportError(field.getName(), exception.message,'RegexValidatorException');
```

Comment:
I think if we want to go this route then the error codes should be defined and exported and referenced.

E.g. something like:

```
ErrorCodes.js
export const INVALID_REGEX = 1;
```

Somewhere we want to report an error:
```
this.reportError(field.getName(), exception.message, ErrorCodes.INVALID_REGEX);
```

This allows a client to do something like:

```
import {ErrorCodes} from '@accordproject/concerto-core';

try {
...
}
catch(err) {
const errorCode = err?.getErrorCode.?();
switch(errorCode) {
case ErrorCodes.INVALID_REGEX:
  // do something specific here
break;
default:
  // general case
}
```

We should have in mind that concerto-core will be moving to TypeScript and we want these error codes to be type safe.

Reply:
Hi @dselman , as you requested I have changed the structure by doing the following changes to the structure:
1. I no longer am using the StringValidatorException class instead I have added the errorType as an optional param in BaseException and using that.
2. added an errorCodes.js file where I am storing the error type constants and using them directly , these are exposed so the client will also be able to use these , as discussed in the PR